firmware/spiders/linksys.py

```diff
@@ -0,0 +1,111 @@
+from scrapy import Request, Spider
```

Comment:
Refactor import order.
PEP style is: built ins first (i.e. re, typing), third party second (i.e. scrapy), first party third (i.e. firmware.items).
Then alphabetical in each block plus a blank line dividing the blocks